crates/wdk-build/src/cargo_make.rs

```diff
@@ -548,11 +566,16 @@ pub fn setup_path() -> Result<impl IntoIterator<Item = String>, ConfigError> {
     );
 
     let wdk_tool_root = get_wdk_tools_root(&wdk_content_root, sdk_version);
-    let host_windows_sdk_version_tool_path =
-        absolute(wdk_tool_root.join(host_arch.as_windows_str()))?
-            .to_str()
-            .expect("WDK tool path should be valid UTF-8")
-            .to_string();
+    let host_windows_sdk_version_tool_path = {
+        let path = wdk_tool_root.join(host_arch.as_windows_str());
+        absolute(&path).map_err(|source| IoError {
+            metadata: IoErrorMetadata::SinglePath { path },
+            source,
+        })
```

Comment:
`IoError` could use a constructor that takes `source` and `path` (e.g. `IoError::with_path(path, source)`) to remove repeated verbose initialization.